pkg/daemon/ceph/osd/daemon.go

```diff
@@ -328,6 +328,14 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 				logger.Errorf("failed to get udev info of partition %q. %v", device.Name, err)
 				continue
 			}
+			if device.PartitionTableType != sys.GPTType {
+				pType := "unknown or untyped (possibly atari)"
+				if device.PartitionTableType != "" {
+					pType = device.PartitionTableType
+				}
+				logger.Infof("skipping partition %q because it is not a gpt partition. detected: %q", device.Name, pType)
+				continue
+			}
```

Comment:
IMPORTANT NOTE: if the partition is atari, `udevadm info --query=property` lists the type as being the same as the partition table type of the parent disk. If the parent is given a `gpt` scheme but there is an atari partition, `udevadm` reports that the partition is `gpt` erroneously!

@leseb @travisn